x-pack/plugins/uptime/server/lib/synthetics_service/get_api_key.ts

```diff
@@ -93,3 +97,60 @@ export const generateAndSaveAPIKey = async ({
     return apiKeyObject;
   }
 };
+
+export const deleteServiceApiKey = async ({
+  savedObjectsClient,
+}: {
+  savedObjectsClient: SavedObjectsClientContract;
+}) => {
+  await deleteSyntheticsServiceApiKey(savedObjectsClient);
+};
+
+export const canEnableSynthetics = async ({
+  request,
+  server: { uptimeEsClient, security, encryptedSavedObjects },
+}: {
+  server: UptimeServerSetup;
+  request?: KibanaRequest;
+}) => {
+  const encryptedClient = encryptedSavedObjects.getClient({
+    includedHiddenTypes: [syntheticsServiceApiKey.name],
+  });
+
+  const [apiKey, hasPrivileges, areApiKeysEnabled] = await Promise.all([
+    getSyntheticsServiceAPIKey(encryptedClient),
+    uptimeEsClient.baseESClient.security.hasPrivileges({
+      body: {
+        cluster: [
+          'manage_security',
+          'manage_api_key',
+          'manage_own_api_key',
+          ...serviceApiKeyPrivileges.cluster,
+        ],
+        index: serviceApiKeyPrivileges.index,
+      },
+    }),
+    security.authc.apiKeys.areAPIKeysEnabled(),
+  ]);
+
+  const {
+    cluster: {
+      manage_security: manageSecurity,
+      manage_api_key: manageApiKey,
+      manage_own_api_key: manageOwnApiKey,
+      monitor,
+      read_ilm: readILM,
+      read_pipeline: readPipeline,
+    },
+  } = hasPrivileges;
+
+  const canManageApiKeys = manageSecurity || manageApiKey || manageOwnApiKey;
```

Comment:
i think this should be && ? i mean if they can't manage security, they can't access anything... AFAIK

Reply:
`manage_security` encompasses all security CRUD operations, including API keys.